### PR TITLE
Weather App v1.1

### PR DESCRIPTION
**Pull request by ybadmus**
The Weather App project comes at the end of the Asynchronous Javascript and APIs section of the Javascript course. It tests the knowledge of the student about asynchronous communication with promises or async/await and when to use them.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      loadResults(weather);
+    }
+  });
+};
+
+const main = () => {
+  document.body.style.background = `url(${darkNight})`;
+
+  const mainView = document.getElementById('mainView');
+  mainView.style.display = 'flex';
```
Comment (rindrajosia) [inline src/index.js]: Keep all the styling logic in CSS files

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+  const resultView = document.getElementById('resultView');
+  resultView.style.display = 'none';
```
Comment (rindrajosia) [inline src/index.js]: Keep all the styling logic in CSS files

File: src/modules/helper.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  document.getElementById('result-feeling').textContent = '';
+  document.getElementById('result-humidity').textContent = '';
+  document.getElementById('result-pressure').textContent = '';
+  document.getElementById('result-mintemp').textContent = '';
+  document.getElementById('result-maxtemp').textContent = '';
+};
+
+const toggleMainView = () => {
+  const mainView = document.getElementById('mainView');
+  mainView.style.display = 'none';
```
Comment (rindrajosia) [inline src/modules/helper.js]: Keep all the styling logic in CSS files

File: src/modules/helper.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+  const resultView = document.getElementById('resultView');
+  resultView.style.display = 'block';
```
Comment (rindrajosia) [inline src/modules/helper.js]: Keep all the styling logic in CSS files

File: src/modules/helper.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+};
+
+const toggleDetailsView = () => {
+  const mainView = document.getElementById('mainView');
+  mainView.style.display = 'block';
```
Comment (rindrajosia) [inline src/modules/helper.js]: Keep all the styling logic in CSS files

File: src/modules/helper.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+  const resultView = document.getElementById('resultView');
+  resultView.style.display = 'none';
```
Comment (rindrajosia) [inline src/modules/helper.js]: Keep all the styling logic in CSS files

File: src/index.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+      }
+
+      localStorage.setItem('weather-unit', 'fahrenheit');
+      loadResults(weather);
+    }
+  });
+};
+
+const main = () => {
+  document.body.style.background = `url(${darkNight})`;
```
Comment (rindrajosia) [inline src/index.js]: Keep all the styling logic in CSS files

File: README.md [hunk 8]
```diff
@@ -1,2 +1,63 @@
-# weather-app
+
+
+## To-do List
```
Comment (rindrajosia) [inline README.md]: Please, replace this to weather-app

File: README.md [hunk 9]
```diff
@@ -1,2 +1,63 @@
...
+![screenshot](./screenshot.png)
+
+## Live Link
+Visit the [To-do List](http://www.ybadmus.me/weather-app/) to get started
```
Comment (rindrajosia) [inline README.md]: Please, replace To-do List to weather-app.

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- Webpack is well configured :+1: 
- Good use of ES6 syntax :100: 

To improve :factory_worker: [REQUIRED]:

- [x] It changes its appearance based on the received data ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/appearance_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app))
- [x] Keep all the styling logic in CSS files (If you need to change something like a background image, add the classes with the different backgrounds to the CSS and then change the class name attribute with JavaScript) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/styling_requirement.png) from Pathwright)
- [x] Please make sure that the information in your readme is correct (check inline comments below :arrow_double_down:), and don't forget to include all that's needed for running the app with the API(s) it consumes. It should also have instructions about where and how to generate any necessary keys, credentials, how to set, store, and generate them.
- [x] I got an error (Uncaught (in promise) TypeError: weather.main is undefined) when I search a location on my local machine, it will be nice if you fix it:


- [x] I suggest that your APP displays a message when the location that the user entered doesn't exist.

Nice to have [OPTIONAL]

- [ ]  it fetches the current geolocation of the user when the app is first accessed and fetches the weather from that location

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

**Review by mahmoud717 — CHANGES_REQUESTED** [review]
### Changes required ♻️ 

**Hello @ybadmus**

Great job with the project. you have implemented most of the required changes, but there are still some changes to be made:
 
---
### Requested 🔴 

- [ ] Please make sure that the app changes background based on the results. you can change it from day to night or the other way around. 

- [ ] Please make sure to add error handling for when the user enters an invalid city name. right now it just throws a console error.

---

please make these changes and request another code review.

if you have any questions don't hesitate to contact me on slack **MahmoudMohammad**

**Comment by ybadmus** [comment]
It looks like @mahmoud717 didn't do any review at all. All the issues he raised have been fixed so I'm surprised he is reporting the same things. Not fair man!

**Review by Temmarie — APPROVED** [review]
## Status: Approved :heavy_check_mark:

### Hi @ybadmus ! :wave:

#### Thank you for making all of the required changes, your project is approved :clap: :clap: :clap: :tada:
 
#### You may merge, good luck with your next projects! :rocket:
#### Happy Coding! :+1:
__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

**Pull request merged**